ionoscloud/resource_server.go

```diff
@@ -767,9 +764,12 @@ func resourceServerCreate(ctx context.Context, d *schema.ResourceData, meta inte
 						return diags
 					}
 				}
-
-				volumeItems := serverReq.Entities.Volumes.Items
-				firstVolumeItem := (*volumeItems)[0]
+				var volumeItems *[]ionoscloud.Volume
+				var firstVolumeItem ionoscloud.Volume
+				if serverReq.Entities.Volumes != nil {
+					volumeItems = serverReq.Entities.Volumes.Items
```

Comment:
@cristiGuranIonos shouldn't we also check if `serverReq.Entities.Volumes.Items != nil` and `serverReq.Entities.Volumes.Items` is not an empty list?

Reply:
added a check below before instantiating the first volume. if volumes is != nil then most likely items is populated